packages/@aws-cdk/aws-stepfunctions-tasks/lib/athena/start-query-execution.ts

```diff
@@ -0,0 +1,289 @@
+            KmsKey: this.props.resultConfiguration?.encryptionConfiguration?.encryptionKey,
+          },
+          OutputLocation: this.props.resultConfiguration?.outputLocation,
+        },
+        WorkGroup: this.props.workGroup,
+      }),
+    };
+  }
+}
+
+/**
+ * Location of query result along with S3 bucket configuration
+ *
+ * @see https://docs.aws.amazon.com/athena/latest/APIReference/API_ResultConfiguration.html
+ * @experimental
+ */
+export interface ResultConfiguration {
+
+  /**
+   * S3 path of query results
+   *
+   * @default - Query Result Location set in Athena settings for this workgroup
+   * @example s3://query-results-bucket/folder/
+  */
+  readonly outputLocation?: string;
+
+  /**
+   * Encryption option used if enabled in S3
+   *
+   * @default - SSE_S3 encrpytion is enabled with default encryption key
+   */
+  readonly encryptionConfiguration?: EncryptionConfiguration
+}
+
+/**
+ * Encryption Configuration of the S3 bucket
+ *
+ * @see https://docs.aws.amazon.com/athena/latest/APIReference/API_EncryptionConfiguration.html
+ * @experimental
+ */
+export interface EncryptionConfiguration {
+
+  /**
+   * Type of S3 server-side encryption enabled
+   *
+   * @default - EncryptionOption.S3_MANAGED
+   */
+  readonly encryptionOption: EncryptionOption;
+
+  /**
+   * KMS key ARN or ID
+   *
+   * @default - No KMS key for Encryption Option SSE_S3 and default master key for Encryption Option SSE_KMS and CSE_KMS
+   */
+  readonly encryptionKey?: kms.IKey;
+}
+
+/**
+ * Encryption Options of the S3 bucket
+ *
+ * @see https://docs.aws.amazon.com/athena/latest/APIReference/API_EncryptionConfiguration.html#athena-Type-EncryptionConfiguration-EncryptionOption
+ * @experimental
+ */
+export enum EncryptionOption {
+  /**
+   * Server side encryption (SSE) with an Amazon S3-managed key.
+   *
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html
+   */
+  S3_MANAGED = 'SSE_S3',
+
+  /**
+   * Server-side encryption (SSE) with an AWS KMS key managed by the account owner.
+   *
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html
+   */
+  KMS = 'SSE_KMS',
+
+  /**
+   * Client-side encryption (CSE) with an AWS KMS key managed by the account owner.
+   *
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingClientSideEncryption.html
+   */
+  CLIENT_SIDE_KMS = 'CSE_KMS'
+}
+
+/**
+ * Database and data catalog context in which the query execution occurs
+ *
+ * @see https://docs.aws.amazon.com/athena/latest/APIReference/API_QueryExecutionContext.html
+ * @experimental
+ */
+export interface QueryExecutionContext {
+
+  /**
+   * Name of catalog used in query execution
+   *
+   * @default - No catalog
+   */
+  readonly catalogName?: string;
+
+  /**
+   * Name of database used in query execution
+   *
+   * @default - No database
+   */
+  readonly databaseName?: string;
```

Comment:
curious: is a query execution context valid if neither is present?

Reply:
I am going to provide two use cases and then hopefully that will clarify things:
relevant link https://docs.aws.amazon.com/IAM/latest/UserGuide/list_awsglue.html

1. Start an an Athena Query that creates a table, this requires write permissions to catalog, database and table as shown in the actions table in the link above.
2. Start an Athena Query that gets a userdefinedfunction(UDF) then runs the function. This requires read permissions to a UDF

In the table Resource types defined by AWS Glue the table shows how to scope down permissions if one has the UDF or table name which I don’t.

Basically any athena query possible must be able to run from a state machine hence the large permissions granted. 

One thing I can do is that in the table Resource types defined by AWS Glue, the permissions are scoped down if the user sends an optional parameter databaseName.
i.e. currently: UDF /glue/user-defined-function/*
changed to: /glue/user-defined-function/DATABASENAME/*
or if no database name provided /glue/user-defined-function/default/*

This will prevent cross database access.

Reply:
got it, thanks for clarifying. I think we should seek to simplify this. it's unfortunate we don't have L2 constructs for Athena yet, as `grant*` APIs would/should be leveraged here. 

let's introduce any changes that will ensure least privilege for now.

Reply:
I believe those changes have been included in the commit found at these two links:

https://github.com/aws/aws-cdk/pull/11045/files#diff-edd80b0b061b6b337000af6848daeda4011822de72da5749dbcd430c8089bc37R149
https://github.com/aws/aws-cdk/pull/11045/files#diff-edd80b0b061b6b337000af6848daeda4011822de72da5749dbcd430c8089bc37R154

Granting access to all tables and user defined functions in the specified database or the default database if no database has been included in the request.